GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -101,18 +106,21 @@ public partial class GitIndexProjection : IDisposable, IProfilerOnlyIndexProject
             this.fileSystemVirtualizer = fileSystemVirtualizer;
             this.indexParser = new GitIndexParser(this);
 
-            this.projectionReadWriteLock = new ReaderWriterLockSlim();
-            this.projectionParseComplete = new ManualResetEventSlim(initialState: false);
-            this.wakeUpIndexParsingThread = new AutoResetEvent(initialState: false);
             this.projectionIndexBackupPath = Path.Combine(this.context.Enlistment.DotGVFSRoot, ProjectionIndexBackupName);
             this.indexPath = Path.Combine(this.context.Enlistment.WorkingDirectoryBackingRoot, GVFSConstants.DotGit.Index);
             this.placeholderDatabase = placeholderDatabase;
+            this.includedFolderCollection = includedFolderCollection;
             this.modifiedPaths = modifiedPaths;
+            this.rootIncludedFolder = new IncludedFolderData();
+            this.ClearProjectionCaches();
         }
 
-        // For Unit Testing
         protected GitIndexProjection()
         {
+            this.projectionReadWriteLock = new ReaderWriterLockSlim();
+            this.projectionParseComplete = new ManualResetEventSlim(initialState: false);
+            this.wakeUpIndexParsingThread = new AutoResetEvent(initialState: false);
+            this.rootFolderData.ResetData(new LazyUTF8String("<root>"));

```

Comment:
Why did this line need to be added as part of these changes?

`this.rootFolderData.ResetData(new LazyUTF8String("<root>"));`

Reply:
I thought that I made this change for unit tests that were getting a `NullReferenceException` but maybe I fixed that with some other change because it is no longer happening.